### PR TITLE
Migrate System Tests to Chai

### PR DESCRIPTION
**Pull request by codenirvana**
- Use chai expect assertions
- Add global expect for system test

### CODE REVIEW

File: test/system/repository.test.js [hunk 1]
```diff
@@ -103,22 +101,22 @@ describe('project repository', function () {
                     fs.readFile(name, function (error, content) {
                         expect(error).to.equal(null);
                         if (fileDetails.ext === '.sh') {
-                            expect(/^#!\/bin\/bash[\r\n][\W\w]*$/g.test(content)).to.be.ok();
+                            expect(/^#!\/bin\/bash[\r\n][\W\w]*$/g.test(content)).to.be.ok;
```
Comment (kunagpal) [inline test/system/repository.test.js]: Here as well

File: test/system/repository.test.js [hunk 2]
```diff
@@ -184,57 +182,57 @@ describe('project repository', function () {
...
         describe(gitignorePath, function () {
-            it('must exist', function (done) {
+            it('should exist', function (done) {
                 fs.stat(gitignorePath, done);
             });
 
-            it('must have valid content', function () {
-                expect(_.isEmpty(gitignore)).to.not.be.ok();
+            it('should have valid content', function () {
+                expect(_.isEmpty(gitignore)).to.not.be.ok;
```
Comment (kunagpal) [inline test/system/repository.test.js]: `expect(gitignore).to.not.be.empty;`

File: test/system/repository.test.js [hunk 3]
```diff
@@ -184,57 +182,57 @@ describe('project repository', function () {
...
-            it('must have valid content', function () {
-                expect(_.isEmpty(npmignore)).to.not.be.ok();
+            it('should have valid content', function () {
+                expect(_.isEmpty(npmignore)).to.not.be.ok;
             });
         });
 
-        it('.gitignore coverage must be a subset of .npmignore coverage', function () {
+        it('should have .gitignore coverage to be a subset of .npmignore coverage', function () {
             expect(_.intersection(gitignore, npmignore)).to.eql(gitignore);
```
Comment (kunagpal) [inline test/system/repository.test.js]: Assertion message needed

File: test/system/repository.test.js [hunk 4]
```diff
@@ -184,57 +182,57 @@ describe('project repository', function () {
...
     describe('CHANGELOG.yaml', function () {
-        it('must exist', function (done) {
+        it('should exist', function (done) {
             fs.stat('./CHANGELOG.yaml', done);
         });
 
-        it('must have readable content', function () {
-            expect(yml.safeLoad(fs.readFileSync('./CHANGELOG.yaml'))).to.be.ok();
+        it('should have readable content', function () {
+            expect(yml.safeLoad(fs.readFileSync('./CHANGELOG.yaml'))).to.be.ok;
```
Comment (kunagpal) [inline test/system/repository.test.js]: Assertion message needed

File: test/system/appveyor-yml.test.js [hunk 5]
```diff
@@ -15,20 +14,20 @@ describe('appveyor.yml', function () {
...
-        expect(appveyorYAMLError && appveyorYAMLError.message || appveyorYAMLError).to.not.be.ok();
+    it('should be a valid yml', function () {
+        expect(appveyorYAMLError && appveyorYAMLError.message || appveyorYAMLError).to.be.undefined;
     });
 
     describe('strucure', function () {
-        it('init script must exist', function () {
-            expect(appveyorYAML.init[0]).to.be('git config --global core.autocrlf input');
+        it('should have init script', function () {
+            expect(appveyorYAML.init[0]).to.equal('git config --global core.autocrlf input');
```
Comment (shamasis) [inline test/system/appveyor-yml.test.js]: Assertion could be chained for better chai usage using yaml .to.have.property(‘init’).and.to.have …

File: test/system/appveyor-yml.test.js [hunk 6]
```diff
@@ -46,20 +45,21 @@ describe('appveyor.yml', function () {
         });
 
         it('should have a valid install pipeline', function () {
-            expect(appveyorYAML.install[0].ps).to.be('Install-Product node $env:nodejs_version');
-            expect(appveyorYAML.install[1]).to.be('npm cache clean --force');
-            expect(appveyorYAML.install[2]).to.be('appveyor-retry npm install');
+            expect(appveyorYAML.install[0].ps).to.equal('Install-Product node $env:nodejs_version');
```
Comment (shamasis) [inline test/system/appveyor-yml.test.js]: Deep assertion should have been better.

File: test/system/appveyor-yml.test.js [hunk 7]
```diff
@@ -46,20 +45,21 @@ describe('appveyor.yml', function () {
...
+            expect(appveyorYAML.install[1]).to.equal('npm cache clean --force');
+            expect(appveyorYAML.install[2]).to.equal('appveyor-retry npm install');
         });
 
-        it('MS build script and deploy must be turned off', function () {
-            expect(appveyorYAML.build).to.be('off');
-            expect(appveyorYAML.deploy).to.be('off');
+        it('should have MS build script and deploy to be turned off', function () {
+            expect(appveyorYAML.build).to.equal('off');
```
Comment (shamasis) [inline test/system/appveyor-yml.test.js]: .to.have.property(‘build’, ‘off’)

File: test/system/appveyor-yml.test.js [hunk 8]
```diff
@@ -46,20 +45,21 @@ describe('appveyor.yml', function () {
...
+            expect(appveyorYAML.deploy).to.equal('off');
         });
 
-        it('notifications must be configured correctly', function () {
-            expect(appveyorYAML.notifications).to.be.an(Array);
-            expect(appveyorYAML.notifications[0].provider).to.be('Slack');
-            expect(appveyorYAML.notifications[0].incoming_webhook.secure).to.be.ok();
+        it('should have notifications configured correctly', function () {
+            expect(appveyorYAML.notifications).to.be.an('array');
```
Comment (shamasis) [inline test/system/appveyor-yml.test.js]: Deep assertion would have been better.

**Pull request merged**